### PR TITLE
[IMP] stock: when picking is canceled unlock button should not displayed

### PR DESCRIPTION
**Pull request by vas-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/views/stock_picking_views.xml [hunk 1]
```diff
@@ -217,7 +217,7 @@
...
                     <button name="button_scrap" type="object" string="Scrap" attrs="{'invisible': ['|', '|', '&amp;', ('picking_type_code', '!=', 'incoming'), ('state', 'in', ('draft', 'cancel', 'waiting')), '&amp;', ('picking_type_code', '=', 'incoming'), ('state', '!=', 'done'), ('is_locked', '=', False)]}"/>
-                    <button name="action_toggle_is_locked" attrs="{'invisible': ['|', ('state', '=', 'draft'), ('is_locked', '=', False)]}" string="Unlock" groups="stock.group_stock_manager" type="object" help="If the picking is unlocked you can edit initial demand (for a draft picking) or done quantities (for a done picking)."/>
+                    <button name="action_toggle_is_locked" attrs="{'invisible': ['|', ('state', 'in', ('draft','cancel')), ('is_locked', '=', False)]}" string="Unlock" groups="stock.group_stock_manager" type="object" help="If the picking is unlocked you can edit initial demand (for a draft picking) or done quantities (for a done picking)."/>
                     <button name="action_toggle_is_locked" attrs="{'invisible': [('is_locked', '=', True)]}" string="Lock" class="oe_highlight" groups="stock.group_stock_manager" type="object"/>
```
Comment (jco-odoo) [inline addons/stock/views/stock_picking_views.xml]: Should not we lock it automatically then when we cancel?

Comment (jco-odoo) [reply addons/stock/views/stock_picking_views.xml]: self.is_locked = True in action_cancel of mo

Comment (jir-odoo) [reply addons/stock/views/stock_picking_views.xml]: Josse , It is for stock picking only we have not done it for mo. Purpose is to hide unlock button on cancel state of picking. 

Yes we can set it as locked when we canceling mo.

**Comment by jco-odoo** [comment]
It would be good to have the same behaviour for picking as mo.

File: addons/stock/views/stock_picking_views.xml (hunk 1 above)
Comment (jco-odoo) [reply addons/stock/views/stock_picking_views.xml]: I added a commit. Please check.

**Pull request merged**